mollie/api/client.py

```diff
@@ -179,25 +187,81 @@ def perform_http_call(self, http_method, path, data=None, params=None):
             url += '?' + querystring
             params = None
 
-        try:
-            response = requests.request(
-                http_method, url,
-                verify=True,
-                headers={
-                    'Accept': 'application/json',
-                    'Authorization': 'Bearer {api_key}'.format(api_key=self.api_key),
-                    'Content-Type': 'application/json',
-                    'User-Agent': self.user_agent,
-                    'X-Mollie-Client-Info': self.UNAME,
-                },
-                params=params,
-                data=data,
-                timeout=self.timeout,
-            )
-        except Exception as err:
-            raise RequestError('Unable to communicate with Mollie: {error}'.format(error=err))
+        if self.oauth:
+            try:
+                response = self.oauth.request(
+                    http_method,
+                    url,
+                    params=params,
+                    data=data,
+                )
+            except Exception as err:
+                raise RequestError('Unable to communicate with Mollie: {error}'.format(error=err))
+        else:
+            try:
+                response = requests.request(
+                    http_method, url,
+                    verify=True,
+                    headers={
+                        'Accept': 'application/json',
+                        'Authorization': 'Bearer {api_key}'.format(api_key=self.api_key),
+                        'Content-Type': 'application/json',
+                        'User-Agent': self.user_agent,
+                        'X-Mollie-Client-Info': self.UNAME,
+                    },
+                    params=params,
+                    data=data,
+                    timeout=self.timeout,
+                )
+            except Exception as err:
+                raise RequestError('Unable to communicate with Mollie: {error}'.format(error=err))
         return response
 
+    def setup_oauth(self, client_id, client_secret, redirect_uri, scope, token, set_token):
```

Comment:
This method should do a call to `set_user_agent_component`, just like `set_access_token` does.

Reply:
Done.